tasty-plutus/test/Properties/Main.hs

```diff
@@ -0,0 +1,172 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Main (main) where
+
+--------------------------------------------------------------------------------
+
+import Prelude hiding (fmap, ($), (&&), (+), (-), (==))
+
+import Ledger.Crypto (PubKeyHash)
+import Plutus.V1.Ledger.Value (Value)
+import Test.QuickCheck.Plutus.Instances ()
+import Test.Tasty (TestTree, defaultMain, localOption)
+import Test.Tasty.Plutus.Context (
+  ContextBuilder,
+  Purpose (ForSpending),
+  paysToPubKey,
+  paysToSelf,
+ )
+import Test.Tasty.Plutus.Script.Property (scriptProperty)
+import Test.Tasty.Plutus.TestData (
+  Generator (GenForSpending),
+  Methodology (Methodology),
+  Outcome (Fail, Pass),
+  TestItems (
+    ItemsForSpending,
+    spendCB,
+    spendDatum,
+    spendOutcome,
+    spendRedeemer,
+    spendValue
+  ),
+ )
+import Test.Tasty.Plutus.WithScript (toTestValidator, withValidator)
+import Test.Tasty.QuickCheck (Gen, QuickCheckTests (QuickCheckTests), arbitrary, genericShrink, oneof)
+
+--------------------------------------------------------------------------------
+
+import Plutus.V1.Ledger.Contexts (ScriptContext, TxOut)
+import Plutus.V1.Ledger.Contexts qualified as Contexts
+import Plutus.V1.Ledger.Contexts.Extra qualified as Contexts.Extra
+import Plutus.V1.Ledger.Scripts (
+  Validator,
+  mkValidatorScript,
+ )
+import Plutus.V1.Ledger.Value qualified as Value
+import PlutusTx (BuiltinData, applyCode)
+import PlutusTx.Prelude (fmap, fromMaybe, trace, traceIfFalse, uniqueElement, zero, ($), (&&), (-), (==))
+import PlutusTx.TH (compile)
+import Wallet.Emulator.Types (WalletNumber (WalletNumber))
+import Wallet.Emulator.Wallet (Wallet, fromWalletNumber, walletPubKeyHash)
+
+--------------------------------------------------------------------------------
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests =
+  localOption (QuickCheckTests 100) $
+    withValidator "Property based testing" testSimpleValidator $ do
+      scriptProperty "Output secret" $ GenForSpending gen1 transform1
```

Comment:
It is a little bit unclear for me what property this transform test.

Reply:
Hmm. Yes, my validator example is a little weird. So I built a new, simpler one.